src/gc/gc.cpp

```diff
@@ -5830,13 +5837,27 @@ void void_allocation (gc_alloc_context* acontext, void*)
 
 void gc_heap::repair_allocation_contexts (BOOL repair_p)
 {
-    GCToEEInterface::GcEnumAllocContexts (repair_p ? repair_allocation : void_allocation, NULL);
+    alloc_context *acontext;
+    if (!using_thread_alloc_contexts())
```

Comment:
>if (!using_thread_alloc_contexts()) [](start = 4, length = 35)

again, you don't need to check here - EnumAllocContexts will enum either the per thread context or the UP context. You do need this line from fix_youngest:

heap_segment_allocated (ephemeral_heap_segment) = alloc_allocated;